packages/wallets/src/features/cashier/modules/Transfer/hooks/useTransferMessages.ts

```diff
@@ -0,0 +1,292 @@
+        transferDirection === 'from' ? availableSumActiveWalletCurrency : availableSumConverted;
+    const targetCurrencyRemainder =
+        transferDirection === 'from' ? availableSumConverted : availableSumActiveWalletCurrency;
+
+    const formattedSourceCurrencyLimit = displayMoney?.(
+        sourceCurrencyLimit,
+        sourceAccount.currencyConfig.display_code,
+        sourceAccount.currencyConfig.fractional_digits
+    );
+    const formattedTargetCurrencyLimit = displayMoney?.(
+        targetCurrencyLimit,
+        targetAccount.currencyConfig.display_code,
+        targetAccount.currencyConfig?.fractional_digits
+    );
+
+    const formattedSourceCurrencyRemainder = displayMoney?.(
+        sourceCurrencyRemainder,
+        sourceAccount.currencyConfig.display_code,
+        sourceAccount.currencyConfig.fractional_digits
+    );
+    const formattedTargetCurrencyRemainder = displayMoney?.(
+        targetCurrencyRemainder,
+        targetAccount.currencyConfig?.display_code,
+        targetAccount.currencyConfig?.fractional_digits
+    );
+
+    if (availableSumActiveWalletCurrency === 0)
+        return {
+            text: `You've reached the lifetime transfer limit from your ${sourceAccount.accountName} to any ${targetWalletType} Wallet. Verify your account to upgrade the limit.`,
+            type: 'error' as const,
+        };
+
+    if (allowedSumActiveWalletCurrency === availableSumActiveWalletCurrency)
+        return {
+            text: `The lifetime transfer limit from ${sourceAccount.accountName} to any ${targetWalletType} Wallet is ${formattedSourceCurrencyLimit} (${formattedTargetCurrencyLimit}).`,
+            type: sourceAmount > sourceCurrencyRemainder ? ('error' as const) : ('success' as const),
+        };
+
+    return {
+        text: `Remaining lifetime transfer limit is ${formattedSourceCurrencyRemainder} (${formattedTargetCurrencyRemainder}). Verify your account to upgrade the limit.`,
+        type: sourceAmount > sourceCurrencyRemainder ? ('error' as const) : ('success' as const),
+    };
+};
+
+const cumulativeAccountLimitsMessageFn = ({
+    displayMoney,
+    exchangeRates,
+    limits,
+    sourceAccount,
+    sourceAmount,
+    targetAccount,
+}: TMessageFnProps) => {
+    const isBetweenWallets = sourceAccount.account_category === 'wallet' && targetAccount.account_category === 'wallet';
+    const isSameCurrency = sourceAccount.currency === targetAccount.currency;
+
+    const keyAccountType =
+        [sourceAccount, targetAccount].find(acc => acc.account_category !== 'wallet')?.account_type ?? 'wallets';
+
+    const platformKey = keyAccountType === 'standard' ? 'dtrade' : keyAccountType;
+
+    //@ts-expect-error needs backend type
+    const allowedSumUSD = limits?.daily_cumulative_amount_transfers?.[platformKey].allowed as number;
+    //@ts-expect-error needs backend type
+    const availableSumUSD = limits?.daily_cumulative_amount_transfers?.[platformKey].available as number;
```

Comment:
The cumulative limits are coming in USD always:

<img width="608" alt="Screenshot 2023-11-30 at 16 51 26" src="https://github.com/binary-com/deriv-app/assets/119863957/34cb9e65-b4f9-461b-86ea-1b0bf3710195">